spec.html

```diff
@@ -34032,7 +34028,7 @@ <h1>
               1. Else,
                 1. Let _ref_ be the substring of _templateRemainder_ from 0 to 1.
                 1. Let _refReplacement_ be _ref_.
-              1. Let _refLength_ be the number of code units in _ref_.
+              1. Let _refLength_ be the length of _ref_.
```

Comment:
We could also eliminate the redundancy of the several «if _templateRemainder_ starts with *"$…"*, then let _ref_ be *"$…"*» steps:
```
1. If _templateRemainder_ starts with *"$$"*, then
  1. Let _advanceBy_ be 2.
  1. Let _replacement_ be *"$"*.
1. Else if _templateRemainder_ starts with *"$`"*, then
  1. Let _advanceBy_ be 2.
  1. Let _replacement_ be the substring of _str_ from 0 to _position_.
1. Else if _templateRemainder_ starts with *"$&amp;"*, then
  1. Let _advanceBy_ be 2.
  1. Let _replacement_ be _matched_.
1. Else if _templateRemainder_ starts with *"$'"* (0x0024 (DOLLAR SIGN) followed by 0x0027 (APOSTROPHE)), then
  1. Let _advanceBy_ be 2.
  …
…
1. Set _templateRemainder_ to the substring of _templateRemainder_ from _advanceBy_.
1. Set _result_ to the string-concatenation of _result_ and _replacement_.
```

Reply:
There's a few different `Let refReplacement be ref.` steps, and it seems like it would be a shame to lose those.